SteamRPC.js

```diff
@@ -110,25 +107,64 @@ async function pollSteamPresence(steamUserId, profiles) {
         gameStatus = "";
         let resJson = await res.json();
         if (resJson.in_game) {
-            gameStatus = `${resJson.in_game.name} (not supported)`;
-            if (resJson.in_game?.logo && resJson.in_game?.rich_presence) {
-                let curr_appid = resJson.in_game.logo.split("/apps/")[1].split("/")[0];
-                let curr_rpc = resJson.in_game.rich_presence;
-                debugLine = "Current Steam RPC Status: " + resJson.in_game.rich_presence;
-                if (profiles.hasOwnProperty(curr_appid)) {
-                    let profile = profiles[curr_appid];
-                    try {
-                        gameStatus = profile.title;
-                        let translatedDiscordRPC = profile.translateSteamPresence(curr_rpc);
-                        if (typeof translatedDiscordRPC !== "object") throw `Profile returned '${typeof translatedDiscordRPC}' instead of an object.`;
-                        discordRPCClient.user?.setActivity(translatedDiscordRPC);
-                    }
-                    catch (codeErr) {
-                        throw new Error(`A code error has occured in the game profile for '${profile.title}'!`, {cause: codeErr});
+            gameStatus = resJson.in_game.name;
+
+            if(resJson.in_game?.rich_presence === "")
+                debugLine = "There is no RPC status for this game.";
+            else {
+                if (resJson.in_game?.logo && resJson.in_game?.rich_presence) {
+                    let currentAppId = resJson.in_game.logo.split("/apps/")[1].split("/")[0];
+                    let currentSteamStatus = resJson.in_game.rich_presence;
```

Comment:
I think the code would be cleaner if it initialized a new clientId variable with the default_client_id app here, and then allows any profile to optionally overwrite the clientId with a game-specific clientId.

At the end of this block of code you'd initialize the discordRPCClient if it wasn't done yet or is current/was set to a different instance, and you'd set the activity. I hope that makes sense.